### PR TITLE
➕ validate styles using JSON Schema

### LINKED ISSUE #3: Try JSONSchema validation instead of PropTypes
* https://github.com/mafintosh/is-my-json-valid
* http://json-schema.org/

Probably it's more powerful, flexible and easier to customize.

I implemented a [silly `onlyShape`](https://github.com/batata-frita/lada/commit/aa092eab9237c49efaaada56f37bdc71159bd84e) with PropTypes to error on any extra-non-declared style props, but it doesn't work with first level of propTypes object keys. This functionality requires [more hacks](https://github.com/airbnb/prop-types-exact/blob/master/src/index.js) 🤷‍♂️

### PR DESCRIPTION
**Pull request by deepsweet**
Closes #3

### CODE REVIEW

File: packages/transpile-web/src/transforms/layout.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { isDef, style } from '../utils'
 
 // TODO: move it away from here to a more abstract package
 // TODO: import from a published package
-import { checkLayoutStyles } from '../../../prop-types/src/'
+import { validateLayoutStyles } from '../../../prop-types/src/'
```
Comment (xaviervia) [inline packages/transpile-web/src/transforms/layout.js]: s/prop-types/validation/

**Comment by xaviervia** [comment]
Looks pretty good. What about `text` schema?

**Comment by xaviervia** [comment]
Oh I just realized it has an `inprogress` flag  🤦‍♂️  :D

**Comment by deepsweet** [comment]
WIP WIP

**Comment by deepsweet** [comment]
@xaviervia just finished text styles validation and catch an error that you forgot to implement `letterSpacing` :stuck_out_tongue:

Unknown styles validation FTW :muscle:

**Pull request merged**